File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        long prev = bank[0].chars().filter(ch -> ch =='1').count();
        long res = 0;
        for(int i=1;i<bank.length;i++){
            long cur = bank[i].chars().filter(ch -> ch =='1').count();
            // long cur =
            if(cur>0){
                res += (cur*prev);
                prev = cur;
            } 
        }
        return (int)res;
    }
}

File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        prev = bank[0].count("1")
        res =0
        for i in range(1,len(bank)):
            cur = bank[i].count("1")
            if cur:
                res+= (cur *prev)
                prev = cur

        return res        
        